src/utils/loose-equal.js

```diff
@@ -16,44 +19,27 @@ function isObject(obj) {
  * Returns boolean true or false
  */
 function looseEqual(a, b) {
-  if (a === b) return true
-  const isObjectA = isObject(a)
-  const isObjectB = isObject(b)
-  if (isObjectA && isObjectB) {
-    try {
-      const isArrayA = isArray(a)
-      const isArrayB = isArray(b)
-      if (isArrayA && isArrayB) {
-        return (
-          a.length === b.length &&
-          a.every((e, i) => {
-            return looseEqual(e, b[i])
-          })
-        )
-      } else if (a instanceof Date && b instanceof Date) {
-        return a.getTime() === b.getTime()
-      } else if (!isArrayA && !isArrayB) {
-        const keysA = keys(a)
-        const keysB = keys(b)
-        return (
-          keysA.length === keysB.length &&
-          keysA.every(key => {
-            return looseEqual(a[key], b[key])
-          })
-        )
-      } else {
-        /* istanbul ignore next */
+  if (a === b) {
+    return true
+  }
+  if (isObject(a) && isObject(b)) {
+    // Using for loop over `Object.keys()` here since some class
+    // keys are not handled correctly otherwise
+    for (const key in a) {
+      const aHasKey = a.hasOwnProperty(key)
+      const bHasKey = b.hasOwnProperty(key)
+      if ((aHasKey && !bHasKey) || (!aHasKey && bHasKey) || !looseEqual(a[key], b[key])) {
         return false
       }
-    } catch (e) {
-      /* istanbul ignore next */
-      return false
     }
-  } else if (!isObjectA && !isObjectB) {
-    return String(a) === String(b)
```

Comment:
We may want to add this before the last `return false` as a catch all comparison.

Reply:
Which exact case should this handle?

Reply:
When there is a case for it we can simply replace the last `return false` with `return String(a) === String(b)`.

Reply:
Possibly a custom object/class/etc that may have a toString method. Vue placed it in their original code for some reason.

Reply:
Yeah, that may work.

We might want to add in a test spec for this util (based on the tests that Vue has for their version of `looseEqual()`

Reply:
Yeah. Still not sure about the need of the `String(a) === String(b)`.